ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -133,6 +141,16 @@ runNodeNetwork registry testBtime numCoreNodes nodeJoinPlan nodeTopology
     -- node and server threads on the head node. These mini protocols begin as
     -- soon as both nodes have joined the network, according to @nodeJoinPlan@.
 
+    -- Do not let the slot advance until all live pipes have been empty for at
+    -- least the @quiescenceThreshold@ duration
+    livePipesVar <- uncheckedNewTVarM LivePipes
+      { nextPipeId = PipeId 0
+      , livePipes  = Map.empty
+      }
+    onSlotChange btime $ \s -> do
+        blockUntilQuiescent livePipesVar quiescenceThreshold
+        atomically $ writeTVar latestDoneSlot s
```

Comment:
My main reasons for now:

  * `Test.Dynamic.General` is not the only call-site for `newTestBlockchainTime`
  * This current approach is pretty awkward, still. I'm not even sure "the slot length" is relevant anymore?
  * So I thought we'd figure it out at this call-site before I attempt to alter a function with multiple call-sites.

Reply:
The only reason I'm thinking of for leaving it in `runTestNetwork` is because it's free to introduce further details (beyond `latestDoneSlot` interactions) that `runNodeNetwork` does not care about, so defining `btime` outside seems right.